app/views/comments/_form.html.erb

```diff
@@ -1,5 +1,6 @@
 <h4>Add a comment</h4>
 <!-- #17 -->
+<% if params[:post_id].present? %>
 <%= form_for [post, comment] do |f| %>
```

Comment:
you could have `@commentable` set in the controller, and treat it the same in the view